src/main/kotlin/com/tschuchort/compiletesting/KotlinCompilation.kt

```diff
@@ -409,10 +414,6 @@ class KotlinCompilation {
 	}
 	/** Performs the 1st and 2nd compilation step to generate stubs and run annotation processors */
 	private fun stubsAndApt(sourceFiles: List<File>): ExitCode {
-		if(annotationProcessors.isEmpty()) {
-			log("No services were given. Not running kapt steps.")
-			return ExitCode.OK
-		}
```

Comment:
I removed this because it was not possible to execute the compiler plugins without also adding an annotation processor.
The problem was that the pluginClasspaths were set after this check.

I restored this and made changes. Inside stubsAndApt, only the annotation processors are added to threadLocalParameters. Then inside compileKotlin, only the compiler plugins are set, so the processors aren't getting executed twice.

Reply:
Wouldn't this change the behaviour so that additional compiler plugins are only registered in the `compileKotlin` step? The current behaviour of #23. Is that additional plugins are executed in both steps. I don't know which way is actually right though,

Reply:
Maybe you are right, we could also do it this way:

```kotlin
if(annotationProcessors.isEmpty() && compilerPlugins.isEmpty()) {
			log("No services were given. Not running kapt steps.")
			return ExitCode.OK
}
```
I remove the changes in stubsAndApt and compileKotlin and just add the additional check for the compilerPlugins. Then the behaviour stays the same and it's possible to add compilerPlugins without also adding annotation processors.

Reply:
I think it would be best to always add compiler plugins in the `stubsAndApt` step but cancel it if there are no annotation processors, like we did previously.

Reply:
I'm not sure if i understand it correctly, in stubsAndApt we'll add the compiler plugins and cancel if there is no annotation processor. And then in compileKotlin we'll add the plugins again?

Otherwise i can't see how it's possible to add a compiler plugin without also having to add a annotation processor.